### ISSUE TITLE
[BUG] Use standard keyboard shortcuts for clearing the search, minimizing the window, or activating the hamburger menu

### ISSUE THREAD

**Issue by nekohayo** [cboxdoerfer/fsearch#392]
Hi there,

Very cool application! I'm impressed by how ludicrously fast it is.
I found some issues with the default keyboard shortcuts. Of course customizable shortcuts (issue #243) are nice, but the defaults should match the expected standard shortcuts found elsewhere.

The "Esc" to minimize key is really weird and unexpected for me, compared to all other applications I've seen with a search field. Typically, `Esc` in a search field is used  for clearing the search field.

In GNOME (and probably elsewhere, I suspect at least in XFCE and others?) the minimize shortcut was `Alt+F9`, though nowadays it has been changed to `Super+H`. I think you could map "Minimize" to both of those shortcuts instead of `Esc`, by default (no need for an option). Or, in fact, I'm not sure it's even the app's responsibility to handle the Minimize shortcut, isn't that the DE & WM's job? Ctrl+H already works to minimize FSearch in GNOME, for example.

Furthermore, `F10` is the default shortcut for opening the hamburger menu (which can then be dismissed with `F10` again or `Esc`), it would be good for accessibility to add that.

Then the Esc key in the search field should be mapped to clearing the contents of the filtering search field.

See also https://wiki.gnome.org/Design/OS/KeyboardShortcuts for a list of default keyboard shortcuts.

**Comment by cboxdoerfer** [comment]
Hi,

> Very cool application! I'm impressed by how ludicrously fast it is.

Thanks, glad you like it!

> I found some issues with the default keyboard shortcuts. Of course customizable shortcuts (issue https://github.com/cboxdoerfer/fsearch/issues/243) are nice, but the defaults should match the expected standard shortcuts found elsewhere.

I agree.

> The "Esc" to minimize key is really weird and unexpected for me, compared to all other applications I've seen with a search field. 

Yes, in hindsight I'm also not a big fan of having that as the default behavior. Not sure though if it makes sense to now add a third option to the Esc handling (nothing, minimize, exit) and make it the default, when the custom shortcuts will make that setting obsolete anyway.

> In GNOME (and probably elsewhere, I suspect at least in XFCE and others?) the minimize shortcut was Alt+F9, though nowadays it has been changed to Super+H. I think you could map "Minimize" to both of those shortcuts instead of Esc, by default (no need for an option). Or, in fact, I'm not sure it's even the app's responsibility to handle the Minimize shortcut, isn't that the DE & WM's job? Super+H already works to minimize FSearch in GNOME, for example.

Yes, I agree. Having the desktop provide a universal minimize shortcut and relying on that by default makes sense. But I definitely want to have the option to bind minimizing to a custom shortcut within the app.

> By freeing this key from the minimize behavior, then the Esc key in the search field should be mapped to clearing the contents of the filtering search field instead.

That's actually something I've never noticed in similar apps. At least not with apps which have a persistent search field. Like most apps usually have popups or window for searching, which then get closed on `<Esc>`, which I guess makes sense. But for persistent search fields like in FSearch I could think of a couple of other intuitive things `<Esc>` could be doing, like cancel the current search or discard any suggestions or auto-completions.

> Furthermore, F10 is the default shortcut for opening the hamburger menu (which can then be dismissed with F10 again or Esc), it would be good for accessibility to add that.

Yes, that's seems like a good default to have.

**Comment by nekohayo** [comment]
I would indeed err on the side of reducing the settings you have in your UI, and just having good defaults for anything that is not controversial or highly performance-impacting. The keyboard shortcuts is one such good example (and the "Esc to exit the app" shortcut doesn't make sense when you can have both `Alt+F4`, `Ctrl+Q`, and to an extent `Ctrl+W`, mapped to closing the app or window(s), by default).

I don't know if you are open to this, but I could have a couple of additional suggestions of "other settings elsewhere in the UI that don't really need to exist", to simplify your code paths and testing matrix (in line with [this long-standing philosophy article](https://ometer.com/preferences.html) that shaped GNOME2 and GNOME3). I could open a ticket for that if you are interested.

**Comment by cboxdoerfer** [comment]
> I would indeed err on the side of reducing the settings you have in your UI, and just having good defaults for anything that is not controversial or highly performance-impacting.

Yes, defaults should be as good/reasonable as possible. I'm also willing to remove or rework unreasonable and (likely) unpopular settings, but this should be discussed on a per-setting basis.

> The keyboard shortcuts is one such good example (and the "Esc to exit the app" shortcut doesn't make sense when you can have both Alt+F4, Ctrl+Q, and to an extent Ctrl+W, mapped to closing the app or window(s), by default), where you could just set the shortcut and not make a dedicated option for it (or in some cases leaving it up to the window manager) until you have a real keyboard shortcuts configuration UI.

I agree, but I think the time to get rid of the dedicated `<Esc>`-to-exit option and alter the default, is when the more universal keyboard shortcuts manager is ready.

> I don't know if you are open to this, but I could have a couple of additional suggestions of "other settings elsewhere in the UI that don't really need to exist", to simplify your code paths and testing matrix (in line with [this long-standing philosophy article](https://ometer.com/preferences.html) that shaped GNOME2, GNOME3, and newer). I could open a ticket for that if you are interested.

Sure, it's always nice to hear an opinion about how things could be improved and I already have a few settings in mind which could be simplified or maybe removed (e.g. by merging two similar settings into one).

But just as an FYI, the trend for this application is to add more settings for still quite a while. Not so much in terms of UI customizability, but to make the search and database more flexible and useful. So the goal for this app is not to be a minimal and highly opinionated search utility, but to be quite flexible and powerful.

**Comment by cboxdoerfer** [comment]
Support for `F10` to open the app menu has been added: bd9272a92f2054ecfd10251c358610c5684a3d3c
